cmd/brew-test-bot.rb

```diff
@@ -977,7 +980,10 @@ def test_ci_upload(tap)
 
     if (pr = ENV["UPSTREAM_PULL_REQUEST"])
       pull_pr = "https://github.com/#{tap.user}/homebrew-#{tap.repo}/pull/#{pr}"
-      safe_system "brew", "pull", "--clean", pull_pr
+      args = []
+      args << "--tap=#{@tap}" unless @tap.nil?
+      args << pull_pr
+      test "brew", "pull", "--clean", *args
```

Comment:
The motivation is to build bottles using `test-bot` whose `INSTALL_RECEIPT.json` has `"tap":"homebrew/science"` rather than `"tap":"linuxbrew/science"` as if the following commands had been run:
```sh
brew tap homebrew/science https://github.com/Linuxbrew/homebrew-science
brew pull --tap=homebrew/science https://github.com/Linuxbrew/homebrew-science/pull/342
brew install --build-bottle bwa
brew bottle bwa
```